packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -72,6 +106,22 @@ export default util.createRule<[Options], MessageIds>({
           return;
         }
 
+        // handling void inside union
+        if (
+          node.parent.type === AST_NODE_TYPES.TSUnionType &&
+          isValidUnionType(node.parent)
+        ) {
+          return;
+        }
+
+        // handling Promise<void> inside union
+        if (
+          node.parent.parent.parent?.type === AST_NODE_TYPES.TSUnionType &&
```

Comment:
triple `parent` might cause issues ? Should we check `node.parent` is a `GenericType` like `Promise<void>` ?